Add Accept Offer modal to Offer pages

delventhalz opened the pull request:
Adds an Accept Offer Modal, which also allows for Holding creation. Also reverse the direction of the bifold arrow in most contexts. The direction of the arrow no reflects the direction of the flow of assets, relative to the user.

_Screenshots:_
![screen shot 2017-12-20 at 1 17 32 pm](https://user-images.githubusercontent.com/8889580/34222138-8a028dca-e588-11e7-9bbc-c28c31f4fce4.png)

Commit:
Add Accept Offer modal to Offer pages

Signed-off-by: Zac Delventhal <zac@bitwise.io>

Commit:
Change direction of bifold arrow

Now goes from right to left by default, but goes left to right when
creating a new offer. This reflects the flow of resources depending
on context.

Signed-off-by: Zac Delventhal <zac@bitwise.io>

boydjohnson commented:
When the AcceptOffer modal opens I get in the console log: TypeError: holdings.find(...) is undefined.

peterschwarz commented on sawbuck_app/src/views/accept_offer_modal.js:
```diff
@@ -0,0 +1,206 @@
+        ])
+      })
+      .then(([ user, owner ]) => {
+        const inAsset = getAsset(vnode.state.offer.source, owner.holdings)
```
Based on feedback from @boydjohnson, is there a condition where the holdings is null?

delventhalz replied:
Shouldn't be. Accounts should be sent back with `"holding": []` if they have no holdings.

Commit:
Remove response bodies from PATCH routes

Sending back proper RESTful responses to these routes introduces
race conditions between the REST API and ledger sync. A proper
solution to this problem will likely require a "big" solution
including possibly some architectural changes. Insead, the RESTful
approach is being abandoned for these two routes, with simply
200 responses sent back instead of full resources.

Signed-off-by: Zac Delventhal <zac@bitwise.io>

Commit:
Make "source" in AcceptOffer spec optional

Signed-off-by: Zac Delventhal <zac@bitwise.io>

boydjohnson reviewed (approved):
It would be nice if instead of "New Holding" it said "New Sawbuck Holding" or "New USD Holding".

boydjohnson commented:
Tested locally and was able to accept offers.

Commit:
Send rules array back within resources even when empty

This matches the expectations of RESTful clients, that arrays will
not be omitted from resources when empty.

Signed-off-by: Zac Delventhal <zac@bitwise.io>
